cumulus_nvue: Manage link when configuring ports

larsks opened the pull request:
Ensure that we bring the link state down when deleting a port (and bring it
back up when configuring a port). These link state changes provide
notifications to connected systems that they should attempt to immediately
re-acquire a DHCP lease. This allows systems to respond quickly when we
move them between networks.

Closes: cci-moc/ops-issues#1530

larsks committed:
cumulus_nvue: Manage link when configuring ports

Ensure that we bring the link state down when deleting a port (and bring it
back up when configuring a port). These link state changes provide
notifications to connected systems that they should attempt to immediately
re-acquire a DHCP lease. This allows systems to respond quickly when we
move them between networks.

Closes: cci-moc/ops-issues#1530

larsks commented:
(This is modeled after the `cumulus` driver.)

hakasapl reviewed (approved):
Looks good to me.

Maybe will cause a merge conflict with https://github.com/CCI-MOC/network-runner/pull/8 depending on what gets merged first

I think this can be deployed directly without any retroactive changes because everything is up at the moment so any new ports will set it redundantly and any ports being deleted will be set to down automatically.

tzumainn commented:
@hakasapl do you want to merge #8 first then?

larsks commented:
I have rebased #8 on [`devel`](https://github.com/CCI-MOC/network-runner/tree/devel), and rebased this PR on #8. So merge #8 first, then merge this one, and everything should apply cleanly.
